analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -122,12 +122,15 @@ private static SymbolReferenceInfo.Types.SymbolReference GetSymbolReference(IRea
             var symbolReference = new SymbolReferenceInfo.Types.SymbolReference { Declaration = GetTextRange(declarationSpan.Value) };
             for (var i = 0; i < references.Count; i++)
             {
-                var reference = references[i];
-                if (!reference.IsDeclaration
-                    && reference.Identifier.GetLocation().GetMappedLineSpanIfAvailable() is var mappedLineSpan
-                    // Syntax tree can contain elements from external files (e.g. razor imports files)
-                    // We need to make sure that we don't count these elements.
-                    && string.Equals(mappedLineSpan.Path, filePath, StringComparison.OrdinalIgnoreCase))
+                if (references[i].IsDeclaration)
+                {
+                    continue;
+                }
+
+                // Syntax tree can contain elements from external files (e.g. razor imports files)
+                // We need to make sure that we don't count these elements.
+                var mappedLineSpan = references[i].Identifier.GetLocation().GetMappedLineSpanIfAvailable();
+                if (string.Equals(mappedLineSpan.Path, filePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
NIT: I don't see how this is better than before, but no need to change it back (except if you want to).

Reply:
As discussed offline: I am reverting the change.